examples/flutter_gallery/lib/demo/material/bottom_app_bar_demo.dart

```diff
@@ -0,0 +1,571 @@
+    const _FabLocationConfiguration('Start, docked to the top app bar', _BabMode.CENTER_FAB, const _TopStartFloatingActionButtonLocation()),
+  ];
+  
+  // The index of the currently-selected _FabShapeConfiguration.
+  int fabShapeIndex = 0;
+
+  final List<_FabShapeConfiguration> _fabShapeConfigurations =  <_FabShapeConfiguration>[
+      const _FabShapeConfiguration('None', null),
+      new _FabShapeConfiguration('Circular', 
+        new Builder(builder: (BuildContext context) {
+          return new FloatingActionButton(
+            onPressed: () => _showSnackbar(context),
+            child: const Icon(Icons.add),
+            backgroundColor: Colors.orange,
+          );
+        }),
+      ),
+      new _FabShapeConfiguration('Diamond',
+        new Builder(builder: (BuildContext context) {
+          return new _DiamondFab(
+            onPressed: () => _showSnackbar(context),
+            child: const Icon(Icons.add),
+          );
+        }),
+      ),
+    ];
+
+  // The currently-selected Color for the Bottom App Bar.
+  Color babColor;
+
+  List<Color> babColors = <Color> [
+    null,
+    Colors.orange,
+    Colors.green,
+    Colors.lightBlue,
+  ];
+
+  // Whether or not to show a notch in the Bottom App Bar around the
+  // Floating Action Button when it is docked.
+  bool notchEnabled = true;
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: const Text('Bottom App Bar with FAB location'), 
+        // Add 48dp of space onto the bottom of the appbar.
+        // This gives space for the top-start location to attach to without
+        // blocking the 'back' button.
+        bottom: const PreferredSize(
+          preferredSize: const Size.fromHeight(48.0), 
+          child: const SizedBox(),
+        ),
+      ),
+      body: new SingleChildScrollView(
+        padding: const EdgeInsets.all(16.0),
+        child: controls(context),
+      ),
+      bottomNavigationBar: new MyBottomAppBar(_fabLocationConfigurations[fabLocationIndex].babMode, babColor, notchEnabled),
+      floatingActionButton: _fabShapeConfigurations[fabShapeIndex].fab,
+      floatingActionButtonLocation: _fabLocationConfigurations[fabLocationIndex].fabLocation,
+    );
+  }
+
+  Widget controls(BuildContext context) {
+    return new Column(
+      children: <Widget> [
+        new Text(
+          'Floating action button',
+          style: Theme.of(context).textTheme.title,
+        ),
+        new Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: <Widget>[
+            new Container(width: 96.0,
+              child: const Text('Shape: '),
+            ),
+            new Expanded(child: buildFabShapePicker()),
+          ],
+        ),
+        new Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: <Widget>[
+            new Container(width: 96.0,
```

Comment:
`width:` on its own line

SizedBox would be a little tider than Container.

Reply:
Done